include/robot_interfaces/robot_backend.hpp

```diff
@@ -188,6 +188,14 @@ class RobotBackend
 
             timers_[3].tic();
             // TODO: this may wait forever
+	    while (!destructor_was_called_ &&
```

Comment:
I think the `!destructor_was_called_` part should be removed. Otherwise there is the possible situation that the deconstructor is called just before this line is executed.  In that case, the while condition would be false and thus the internal check never executed.

Reply:
oh yes, you are right!

Reply:
nice catch!

Reply:
+1, corrected